File: tezocare_mobile/lib/features/medication/domain/entities/medication.dart
import 'package:equatable/equatable.dart';

class Medication extends Equatable {
  final String id;
  final String patientId;
  final String? patientName;
  final String name;
  final String? dosage;
  final String? frequency;
  final String? duration;
  final String? route;
  final DateTime? startDate;
  final DateTime? endDate;
  final String? prescribedBy;
  final String? notes;
  final bool isActive;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  const Medication({
    required this.id,
    required this.patientId,
    this.patientName,
    required this.name,
    this.dosage,
    this.frequency,
    this.duration,
    this.route,
    this.startDate,
    this.endDate,
    this.prescribedBy,
    this.notes,
    required this.isActive,
    this.createdAt,
    this.updatedAt,
  });

  Medication copyWith({
    String? id,
    String? patientId,
    String? patientName,
    String? name,
    String? dosage,
    String? frequency,
    String? duration,
    String? route,
    DateTime? startDate,
    DateTime? endDate,
    String? prescribedBy,
    String? notes,
    bool? isActive,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Medication(
      id: id ?? this.id,
      patientId: patientId ?? this.patientId,
      patientName: patientName ?? this.patientName,
      name: name ?? this.name,
      dosage: dosage ?? this.dosage,
      frequency: frequency ?? this.frequency,
      duration: duration ?? this.duration,
      route: route ?? this.route,
      startDate: startDate ?? this.startDate,
      endDate: endDate ?? this.endDate,
      prescribedBy: prescribedBy ?? this.prescribedBy,
      notes: notes ?? this.notes,
      isActive: isActive ?? this.isActive,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }

  @override
  List<Object?> get props => [
        id,
        patientId,
        patientName,
        name,
        dosage,
        frequency,
        duration,
        route,
        startDate,
        endDate,
        prescribedBy,
        notes,
        isActive,
        createdAt,
        updatedAt,
      ];
}

File: backend/app/models/medication.py
import uuid
from datetime import datetime, timezone, date
from sqlalchemy import Column, String, Boolean, Text, Date, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from app.core.database import Base


class Medication(Base):
    __tablename__ = "medications"

    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    patient_id = Column(UUID(as_uuid=True), ForeignKey("patients.id"), index=True, nullable=False)
    visit_id = Column(UUID(as_uuid=True), ForeignKey("visits.id"), index=True, nullable=False)
    drug_name = Column(String, nullable=False)
    dosage = Column(String, nullable=False)
    frequency = Column(String, nullable=False)
    start_date = Column(Date, nullable=False)
    end_date = Column(Date, nullable=True)
    prescribed_by = Column(String, nullable=True)
    dispensed_by = Column(String, nullable=True)
    is_active = Column(Boolean, default=True, index=True)
    next_refill_date = Column(Date, index=True, nullable=True)
    notes = Column(Text, nullable=True)
    created_at = Column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))

    patient = relationship("Patient", back_populates="medications")
    visit = relationship("Visit", back_populates="medications")
